src/sfizz/Voice.h

```diff
@@ -31,14 +31,38 @@ class Voice {
     /**
      * @brief Construct a new voice with the midistate singleton
      *
+     * @param voiceNumber
      * @param midiState
      */
-    Voice(Resources& resources);
+    Voice(int voiceNumber, Resources& resources);
     enum class TriggerType {
         NoteOn,
         NoteOff,
         CC
     };
+    /**
+     * @brief Get the number which identifies this voice, its index
+     */
+    int getIdNumber() const noexcept
+    {
+        return voiceNumber;
+    }
+
+    enum class State {
+        idle,
+        playing
+    };
+
+    class StateListener {
+    public:
+        virtual void onVoiceStateChanged(int /*idNumber*/, State /*state*/) {}
```

Comment:
On the same idea, the region stores an active/inactive state. Would it be useful to have listeners for this in your applications or a further one?

Reply:
Also thinking some more about it: wouldn't a pointer to `this` be more useful than the id number? or maybe both?

Reply:
The id can be deduced from the Region/Voice, it's true, but on the other hand voice id number creates less coupling, and I don't think I necessarily require my future work to look into details of Voice or Region. (but that remains a thing to be confirmed, I'm not fully sure)